akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphPartitionSpec.scala

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.testkit._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.{ OverflowStrategy, ActorMaterializer, ActorMaterializerSettings, ClosedShape }
+import akka.stream.testkit.Utils._
+import scala.concurrent.duration._
+
+class GraphPartitionSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+    .withInputBuffer(initialSize = 2, maxSize = 16)
+
+  implicit val materializer = ActorMaterializer(settings)
+
+  "A partition" must {
+    import GraphDSL.Implicits._
+
+    "partition to three subscribers" in assertAllStagesStopped {
+      val c1 = TestSubscriber.probe[Int]()
+      val c2 = TestSubscriber.probe[Int]()
+      val c3 = TestSubscriber.probe[Int]()
+
+      RunnableGraph.fromGraph(GraphDSL.create() { implicit b ⇒
+        val partition = b.add(Partition[Int](3, {
+          case g if (g > 3)  ⇒ 0
+          case l if (l < 3)  ⇒ 1
+          case e if (e == 3) ⇒ 2
+        }))
+        Source(List(1, 2, 3, 4, 5)) ~> partition.in
+        partition.out(0) ~> Sink.fromSubscriber(c1)
+        partition.out(1) ~> Sink.fromSubscriber(c2)
+        partition.out(2) ~> Sink.fromSubscriber(c3)
+        ClosedShape
+      }).run()
+
+      c2.request(2)
+      c1.request(2)
+      c3.request(1)
+      c2.expectNext(1)
+      c2.expectNext(2)
+      c3.expectNext(3)
+      c1.expectNext(4)
+      c1.expectNext(5)
```

Comment:
I think you misunderstood that comment. What @rkuhn said is more or less the same as I recommend:
 - for purely functional testing, avoid probes
 - for tests, that *explicitly test* tricky completion issues (like when does a certain port complete relative to another port) which are part of the semantics of the stage, or ordering issues which are also part of the semantics, use probes